src/clj/muuntaja/parse.clj

```diff
@@ -1,21 +1,22 @@
 (ns muuntaja.parse
   (:require [clojure.string :as str])
-  (:import (com.fasterxml.jackson.databind.util LRUMap)))
+  (:import (java.util.concurrent ConcurrentHashMap)))
 
 ;;
-;; Utils
+;; Cache
 ;;
 
-(defn cache [n]
-  (LRUMap. n n))
-
-(defn fast-memoize [^LRUMap cache f]
-  (fn [& args]
-    (or (.get cache args)
-        (let [ret (apply f args)]
-          (when ret
-            (.put cache args ret))
-          ret))))
+(defn fast-memoize [size f]
```

Comment:
I have tried to avoid extra deps like `clojure.core.memoize`, which brings the `core.cache` which has had version problems (maybe, cant recall what). I ran the `lru` tests some year+ ago. Was much faster. I'll rerun to be sure